frontend/src/metabase/entities/bookmarks.js

```diff
@@ -109,11 +109,24 @@ function getEntityFor(type) {
 
 function getIcon(bookmark) {
   const bookmarkEntity = getEntityFor(bookmark.type);
+
+  if (bookmarkEntity.name === "questions") {
+    return bookmarkEntity.objectSelectors.getIcon({
+      ...bookmark,
+      /**
+       * Questions.objectSelectors.getIcon works with Card instances.
+       * In order to reuse it we need to map Bookmark["card_type"] to Card["type"]
+       * because Bookmark["type"] is something else.
+       */
+      type: bookmark.card_type,
+    });
+  }
```

Comment:
I thought it would be simple to refactor but it's a bit tricky because of plugins (in questions and collections) and because the Bookmark does not fully fit into shape of Collection or Card.

For example: `getCollectionIcon` uses `collection.id` to determine the icon, but in bookmarks this information is in `bookmark.item_id` - so there is at least 1 bug there.

I'd rather do this simple change here to keep things working exactly like before, and handle the icon computation in a separate PR.